extensions-core/lucene-extensions/src/main/java/io/druid/lucene/LuceneDirectory.java

```diff
@@ -0,0 +1,19 @@
+package io.druid.lucene;
+
+import io.druid.data.input.InputRow;
+import org.apache.lucene.index.IndexReader;
+import org.apache.lucene.search.IndexSearcher;
+import org.apache.lucene.store.Directory;
+
+import java.io.IOException;
+import java.util.List;
+
+/**
+ */
+public interface LuceneDirectory {
+    public int numRows();
+
+    public IndexReader getIndexReader() throws IOException;
```

Comment:
If user calls IndexReader.close on the returned IndexReader, and then when LuceneDirectory.close() is called, AlreadyClosedException will be thrown.

See comments on class ReadOnlyDirectory